### PR TITLE
Add support to a public access to cloudsql-instance

### PR DESCRIPTION
**Pull request by alefmreis**
- Add support to a public access to cloudsql-instance setting up ipv4_enabled parameter

### CODE REVIEW

File: modules/cloudsql-instance/variables.tf [hunk 1]
```diff
@@ -139,3 +139,9 @@ variable "users" {
   type        = map(string)
   default     = null
 }
+
+variable "ipv4_enabled" {
+  description = "Allow public access to database instance."
```
Comment (juliocc) [inline modules/cloudsql-instance/variables.tf:144]: This adds a public IP to the instance. Can you update the description to reflect that?

**Review by juliocc — APPROVED** [review]
Thanks for the PR! Left you a small comment below.

File: modules/cloudsql-instance/variables.tf (hunk 1 above)
Comment (alefmreis) [reply modules/cloudsql-instance/variables.tf]: Hi @juliocc, yes i do.
Thanks for the help

File: tests/modules/cloudsql_instance/test_plan.py [hunk 2]
```diff
@@ -117,3 +117,26 @@ def test_databases(plan_runner):
...
   assert all(r['values']['instance'] == "db" for r in resources)
   assert sorted(r['values']['name'] for r in resources) == ["db1", "db2"]
+
+
+def test_simple_instance_ipv4_enable(plan_runner):
+  "Test instance ipv4_enabled."
+
+  _, resources = plan_runner(ipv4_enabled="true")
+  assert len(resources) == 1 
+  assert r['values']['settings'][0]['ip_configuration'][0]['ipv4_enabled'] == True
```
Comment (ludoo) [inline tests/modules/cloudsql_instance/test_plan.py:127]:
```python
assert resources[0]['values']['settings'][0]['ip_configuration'][0]['ipv4_enabled']
```

File: tests/modules/cloudsql_instance/test_plan.py [hunk 3]
```diff
@@ -117,3 +117,26 @@ def test_databases(plan_runner):
...
+  "Test replicas ipv4_enabled."
+
+  replicas = """{
+    replica1 = { region = "europe-west3", encryption_key_name = null }
+  }"""
+
+  _, resources = plan_runner(replicas=replicas, ipv4_enabled="true")
+  
+  assert len(resources) == 2
+  r = resources[1]
```
Comment (ludoo) [inline tests/modules/cloudsql_instance/test_plan.py:140]: replace 140-141 with

```python
assert all([r['values']['settings'][0]['ip_configuration'][0]['ipv4_enabled'] for r in resources])
```

File: tests/modules/cloudsql_instance/test_plan.py (hunk 2 above)
Comment (alefmreis) [reply tests/modules/cloudsql_instance/test_plan.py]: done

File: tests/modules/cloudsql_instance/test_plan.py (hunk 3 above)
Comment (alefmreis) [reply tests/modules/cloudsql_instance/test_plan.py]: done, thanks for helping.

**Pull request merged**